### PR TITLE
Add missing tests for the Started state in activity state machine

### PR DESCRIPTION
**Pull request by vitarb**
Added tests for untested transitions and removed some unneeded code from the state machine.

### CODE REVIEW

File: src/machines/activity_state_machine.rs [hunk 1]
```diff
@@ -50,8 +50,7 @@ fsm! {
     Created --(Schedule, on_schedule)--> ScheduleCommandCreated;
 
     ScheduleCommandCreated --(CommandScheduleActivityTask) --> ScheduleCommandCreated;
-    ScheduleCommandCreated
-      --(ActivityTaskScheduled(i64), shared on_activity_task_scheduled) --> ScheduledEventRecorded;
+    ScheduleCommandCreated --(ActivityTaskScheduled(i64), shared on_activity_task_scheduled) --> ScheduledEventRecorded;
```
Comment (Sushisource) [inline src/machines/activity_state_machine.rs:53]: Why change these all back to one line? Enforcing the 100 char wrap is nice, not a big fan of having to scroll around horizontally to read these.

File: src/test_help/canned_histories.rs [hunk 2]
```diff
@@ -567,6 +614,94 @@ pub fn cancel_scheduled_activity_with_signal_and_activity_task_cancel(
...
+/// 7: EVENT_TYPE_WORKFLOW_EXECUTION_SIGNALED
+/// 8: EVENT_TYPE_WORKFLOW_TASK_SCHEDULED
+/// 9: EVENT_TYPE_WORKFLOW_TASK_STARTED
+/// 10: EVENT_TYPE_WORKFLOW_TASK_COMPLETED
+/// 11: EVENT_TYPE_ACTIVITY_TASK_CANCEL_REQUESTED
+/// 12: EVENT_TYPE_WORKFLOW_EXECUTION_SIGNALED
+/// 13: EVENT_TYPE_WORKFLOW_TASK_SCHEDULED
+/// 14: EVENT_TYPE_WORKFLOW_TASK_STARTED
+/// 15: EVENT_TYPE_WORKFLOW_TASK_COMPLETED
+/// 16: EVENT_TYPE_ACTIVITY_TASK_CANCELED
```
Comment (Sushisource) [inline src/test_help/canned_histories.rs:632]: I think maybe you don't need the signal here and can to AT_Canceled right after AT_Cancel_Requested and cover the same path? Not a big deal but might be able to shorten a bit.

Comment (Sushisource) [reply src/test_help/canned_histories.rs:632]: Oh, I guess that's the history of the other one you added. What different paths do they cover? Maybe worth adding a note about that.

**Review by Sushisource — APPROVED** [review]
Just some minor stuff

File: src/test_help/canned_histories.rs (hunk 2 above)
Comment (Sushisource) [reply src/test_help/canned_histories.rs:632]: Which is scheduled vs started? If they cover identical paths maybe we don't need this signal version -- but it sounds like you're saying they cover different paths?

Comment (vitarb) [reply src/test_help/canned_histories.rs:632]: They do cover different paths in the state machine due to activity started event.

**Pull request merged**